src/biokbase/narrative/tests/test_jobmanager.py

```diff
@@ -219,39 +225,216 @@ def test_list_jobs_twice(self):
             jobs_html_1 = self.jm.list_jobs()
             self.assertEqual(jobs_html_0.data, jobs_html_1.data)
 
+    def test_cancel_job__bad_input(self):
+        with self.assertRaisesRegex(ValueError, no_id_err_str):
+            self.jm.cancel_job(None)
+
+        with self.assertRaisesRegex(ValueError, no_id_err_str):
+            self.jm.cancel_job("")
+
+    def test_cancel_jobs__bad_inputs(self):
+        with self.assertRaisesRegex(ValueError, no_id_err_str):
+            self.jm.cancel_jobs()
+
+        with self.assertRaisesRegex(ValueError, no_id_err_str):
+            self.jm.cancel_jobs([])
+
+        with self.assertRaisesRegex(ValueError, no_id_err_str):
+            self.jm.cancel_jobs(["", "", None])
+
+    def test_cancel_job__job_already_finished(self):
```

Comment:
I added more rigorous tests for the `cancel_job` and `cancel_jobs` methods -- before issuing a cancel request to ee2, the method first checks the local completed jobs list and then calls `check_job_canceled` on ee2. If those tests both fail, it sends off the cancel request. Where possible, I have tested which of the ee2 calls is made (by checking whether or not the job manager method that wraps them has been executed) as well as the end result.